src/nft/components/bag/Bag.tsx

```diff
@@ -210,48 +216,119 @@ const Bag = () => {
     !bagIsLocked && setLocked(true)
     setBagStatus(BagStatus.FETCHING_ROUTE)
     try {
-      const data = await queryClient.fetchQuery(['assetsRoute', ethSellObject, itemsToBuy, account], () =>
-        fetchRoute({
-          toSell: [ethSellObject],
-          toBuy: itemsToBuy,
-          senderAddress: account ?? '',
+      if (usingGqlRouting) {
+        fetchGqlRoute({
+          variables: {
+            senderAddress: usingGqlRouting && account ? account : '',
+            nftTrades: usingGqlRouting ? buildNftTradeInputFromBagItems(itemsInBag) : [],
+            tokenTrades: tokenTradeInput ? tokenTradeInput : undefined,
+          },
+          onCompleted: (data) => {
+            if (!data.nftRoute || !data.nftRoute.route) {
+              setBagStatus(BagStatus.ADDING_TO_BAG)
+              setLocked(false)
```

Comment:
fast follow: don't allow removals while signing tx to purchase
![Screen Shot 2023-02-15 at 13 27 30 ](https://user-images.githubusercontent.com/11512321/219171924-910a41f7-d34b-4f6d-920b-5c8f8210aa31.png)